ws-android-sdk/Dockerfile

```diff
@@ -0,0 +1,105 @@
+#FROM ubuntu:14.04
+
+#MAINTAINER Parafuzo <dev@parafuzo.com>
+MAINTAINER Jeremy Ellis <keyfreemusic@gmail.com>
+
+
+
+# Update, upgrade and install packages
+RUN \
+    apt-get update && \
+    apt-get -y install curl unzip python-software-properties software-properties-common lib32stdc++6 lib32z1
+
+# Install Oracle Java JDK
+# https://www.digitalocean.com/community/tutorials/how-to-install-java-on-ubuntu-with-apt-get
+# https://github.com/dockerfile/java/blob/master/oracle-java7/Dockerfile
+RUN \
+    echo oracle-java7-installer shared/accepted-oracle-license-v1-1 select true | debconf-set-selections && \
+    add-apt-repository -y ppa:webupd8team/java && \
+    apt-get update && \
+    apt-get install -y oracle-java7-installer
+
+# Install Android SDK
+# https://developer.android.com/sdk/index.html#Other
+RUN \
+    cd /usr/local/ && \
+    curl -L -O http://dl.google.com/android/android-sdk_r24.3.3-linux.tgz && \
+    tar xf android-sdk_r24.3.3-linux.tgz && \
+    rm android-sdk_r24.3.3-linux.tgz
+
+# Install Android NDK
+# https://developer.android.com/tools/sdk/ndk/index.html
+# https://developer.android.com/ndk/index.html
+RUN \
+    cd /usr/local && \
+    curl -L -O http://dl.google.com/android/ndk/android-ndk-r10e-linux-x86_64.bin && \
+    chmod a+x android-ndk-r10e-linux-x86_64.bin && \
+    ./android-ndk-r10e-linux-x86_64.bin && \
+    rm -f android-ndk-r10e-linux-x86_64.bin
+
+
+# Install Gradle
+RUN cd /usr/local && \
+    curl -L https://services.gradle.org/distributions/gradle-2.5-bin.zip -o gradle-2.5-bin.zip && \
+    unzip gradle-2.5-bin.zip
+
+# Update & Install Android Tools
+# Cloud message, billing, licensing, play services, admob, analytics
+RUN \
+    echo y | /usr/local/android-sdk-linux/tools/android update sdk --filter tools --no-ui --force -a && \
+    echo y | /usr/local/android-sdk-linux/tools/android update sdk --filter platform-tools --no-ui --force -a && \
+    echo y | /usr/local/android-sdk-linux/tools/android update sdk --filter android-19 --no-ui --force -a && \
+    echo y | /usr/local/android-sdk-linux/tools/android update sdk --filter android-21 --no-ui --force -a && \
+    echo y | /usr/local/android-sdk-linux/tools/android update sdk --filter android-22 --no-ui --force -a && \
+    echo y | /usr/local/android-sdk-linux/tools/android update sdk --filter android-23 --no-ui --force -a && \
+    echo y | /usr/local/android-sdk-linux/tools/android update sdk --filter extra --no-ui --force -a && \
+    echo y | /usr/local/android-sdk-linux/tools/android update sdk --all --filter build-tools-23.0.2 --no-ui --force
+
+# Set PATH
+ENV ANDROID_HOME=/usr/local/android-sdk-linux ANDROID_NDK_HOME=/usr/local/android-ndk-r10e JAVA_HOME=/usr/lib/jvm/java-7-oracle GRADLE_HOME=/usr/local/gradle-2.5
+ENV PATH $PATH:$ANDROID_HOME/tools:$ANDROID_NDK_HOME/platform-tools:$ANDROID_NDK_HOME:$GRADLE_HOME/bin
+
+# Flatten the image
+# https://intercityup.com/blog/downsizing-docker-containers.html
+# Cleaning APT
+RUN \
```

Comment:
Removing files in a seperate layer won't save space at all, it will just layer a tiny (few kilobytes) image on top of the previous one that included all these files. Should be clearing tmp and oracle installer with && on the same `RUN` line